public/display.js

```diff
@@ -1018,35 +1018,55 @@ function displayActivity() {
   if (typeof callback === 'function') {
    callback()
   }
-  lastMessages += news.data.map((newsMessage) =>
-   attachNewsMessage(element, newsMessage)
+  lastMessages.push(
+   ...news.data.map((newsMessage) =>
+    attachNewsMessage(element, newsMessage)
+   )
   )
+  filterAgain()
+  return nextChunk
  }
 
  function clear() {
   element.innerHTML = ''
   hide()
  }
 
- function filter(filterText) {
-  const terms = filterText
+ let filterTerms = []
+ let numFilteredNews = 0
+ async function filter(filterText) {
+  filterTerms = filterText
    .split(/\s+/)
    .map((x) => x.toLowerCase())
+   .filter((x) => x !== '')
+  filterAgain()
+  while (
+   numFilteredNews < 1 &&
+   (await withLoading(loadMore())) > -1
+  ) {
+   // Non-op
+  }
```

Comment:
Need to re-check the scroll position instead since if there's no results it should load more since we are past the end

We would want to load more even if 2 results matched, since there could be more in the next chunk

Extract the scroll callback in main.js into a new function and call it here instead of this while loop:

```
addEventListener('scroll', () => {
 if (scrollY < lastScrollY) {
  body.classList.add('scroll-up')
 } else {
  body.classList.remove('scroll-up')
 }
 ...
```

Reply:
This design currently runs `loadMore()` until a result is found. It then continues to `loadMore()` once you scroll to the bottom. This ensures that it keep searching until it finds a result or the end of the chunk listing.
Try searching for a word at the end of the news or the video I posted "iris."

Reply:
Added a second parameter `scrolledPastBottom(element, exemptZeroScroll = boolean)` to help load more news when the page isn't long enough to have initiated a scroll.